4_arm.ipynb

```diff
@@ -481,11 +836,78 @@
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's use a function called move _wholebody_ ik that summarizes the above processing. The arguments are the 3D position and the rotation angle of the hand.\n",
+    "\n",
+    "```python\n",
+    "import moveit_commander\n",
+    "from geometry_msgs.msg import PoseStamped\n",
+    "\n",
+    "# Specify whole body control as the control target of moveit\n",
+    "whole_body = moveit_commander.MoveGroupCommander (\"whole_body_light\")\n",
+    "whole_body.allow_replanning(True)\n",
+    "whole_body.set_workspace ([-3.0, -3.0, 3.0, 3.0])\n",
+    "\n",
+    "def move_wholebody_ik(x, y, z, roll, pitch, yaw):\n",
+    "    u\"\"\"Function that controls the robot by inverse kinematics of the whole body\n",
+    "\n",
+    "    argument:\n",
+    "        x(float): End effector target x value [m]\n",
+    "        y(float): End effector target y value [m]\n",
+    "        z(float): End effector target z value [m]\n",
+    "        roll(float): End effector target roll value [deg]\n",
+    "        pitch(float): End effector target pitch value [deg]\n",
+    "        yaw(float): End effector target yaw value [deg]\n",
+    "\n",
+    "    Return value:\n",
+    "        True if working correctly, False otherwise\n",
+    "    \"\"\"\n",
+    "\n",
+    "    p = PoseStamped ()\n",
+    "\n",
+    "    # Specify \"map\" coordinates as base coordinate\n",
+    "    p.header.frame_id = \"map\"\n",
+    "\n",
+    "    # Target position of end effector Set the x, y, z\n",
+    "    p.pose.position.x = x\n",
+    "    p.pose.position.y = y\n",
+    "    p.pose.position.z = z\n",
+    "\n",
+    "    # Convert euler angle to a quarternion\n",
+    "    p.pose.orientation = quaternion_from_euler(roll, pitch, yaw)\n",
+    "\n",
+    "    # Set target position and posture\n",
+    "    whole_body.set_pose_target(p)\n",
+    "    return whole_body.go ()\n",
+    "`` ```\n",
+    "\n",
+    "In the function, we convert Euler angles to quaternions. This makes possible to set the direction of the hand by Euler angles that are easy for humans to understand.\n",
+    "\n",
+    "For example, if you want to grab an object from orientation above, set (roll, pitch, yaw)=(180, 0, hand rotation angle).\n",
+    "And from the side (180, 90, hand rotation angle)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "シミュレータ内に物体を出現させて、HSRに拾わせてみましょう。以下のコマンドでどのような物体があるか表示します。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's make an object appear in the simulator and let HSR pick it up. Following command will display what kind of objects exist in the current scene:"
```

Comment:
"Let's make an object appear in the simulator and make the HSR pick it up. The following command will display what kind of objects exist in the current scene:"